build.gradle

```diff
@@ -419,3 +428,280 @@ testingConventions {
+    useCluster testClusters.bwcFollower0
+
+    doFirst {
+        testClusters.bwcLeader0.upgradeNodeAndPluginToNextVersion(replPluginProvider)
+        testClusters.bwcFollower0.upgradeNodeAndPluginToNextVersion(replPluginProvider)
+        getClusters().forEach { cluster ->
+            String alltransportSocketURI = cluster.nodes.stream().flatMap { node ->
+                node.getAllTransportPortURI().stream()
+            }.collect(Collectors.joining(","))
+            String allHttpSocketURI = cluster.nodes.stream().flatMap { node ->
+                node.getAllHttpSocketURI().stream()
+            }.collect(Collectors.joining(","))
+
+            systemProperty "tests.cluster.${cluster.name}.http_hosts", "${-> allHttpSocketURI}"
+            systemProperty "tests.cluster.${cluster.name}.transport_hosts", "${-> alltransportSocketURI}"
+            systemProperty "tests.cluster.${cluster.name}.security_enabled", "${-> securityEnabled.toString()}"
+            configureCluster(cluster, securityEnabled)
+        }
+    }
+    nonInputProperties.systemProperty('tests.bwcTask', "mixedClusterTask")
+    nonInputProperties.systemProperty('tests.cluster_suffix', "0")
+    filter {
+        setIncludePatterns("org.opensearch.replication.bwc.BackwardsCompatibilityIT")
+    }
+}
+
+// Upgrades the second node to new OpenSearch version with upgraded plugin version after the first node is upgraded.
+// This results in a mixed cluster with 1 node on the old version and 2 upgraded nodes.
+// This is used for rolling upgrade.
+task "twoThirdsUpgradedClusterTask"(type: RestIntegTestTask) {
+    dependsOn "mixedClusterTask"
+    useCluster testClusters.bwcLeader0
+    useCluster testClusters.bwcFollower0
+    doFirst {
+        testClusters.bwcLeader0.upgradeNodeAndPluginToNextVersion(replPluginProvider)
+        testClusters.bwcFollower0.upgradeNodeAndPluginToNextVersion(replPluginProvider)
+        getClusters().forEach { cluster ->
+            String alltransportSocketURI = cluster.nodes.stream().flatMap { node ->
+                node.getAllTransportPortURI().stream()
+            }.collect(Collectors.joining(","))
+            String allHttpSocketURI = cluster.nodes.stream().flatMap { node ->
+                node.getAllHttpSocketURI().stream()
+            }.collect(Collectors.joining(","))
+
+            systemProperty "tests.cluster.${cluster.name}.http_hosts", "${-> allHttpSocketURI}"
+            systemProperty "tests.cluster.${cluster.name}.transport_hosts", "${-> alltransportSocketURI}"
+            systemProperty "tests.cluster.${cluster.name}.security_enabled", "${-> securityEnabled.toString()}"
+            configureCluster(cluster, securityEnabled)
+        }
+    }
+    nonInputProperties.systemProperty('tests.bwcTask', "twoThirdsUpgradedClusterTask")
+    nonInputProperties.systemProperty('tests.cluster_suffix', "0")
+    filter {
+        setIncludePatterns("org.opensearch.replication.bwc.BackwardsCompatibilityIT")
+    }
```

Comment:
can we put this in a common function so that duplication with  `mixedClusterTask` and `rollingUpgradeClusterTask` can be reduced ?

Reply:
I had tried that earlier but there were some issue with setting system properties and thus we had to duplicate the logic.